### PR TITLE
repart: add support for list of definitions directories

### PR DESCRIPTION
**Pull request by rphibel**

### CODE REVIEW

File: src/partition/repart.c [hunk 1]
```diff
@@ -1414,11 +1414,11 @@ static int context_read_definitions(
 
         assert(context);
 
-        dirs = directory ?
-                STRV_MAKE_CONST(directory) :
+        dirs = directories ?
+                directories :
```
Comment (poettering) [inline src/partition/repart.c]: use the gnu short form:

```c
dirs = directories ?: (const char* const*)CONF_PATHS_STRV("repart.d");
```

File: src/partition/repart.c [hunk 2]
```diff
@@ -4281,11 +4281,15 @@ static int parse_argv(int argc, char *argv[]) {
...
 
-                case ARG_DEFINITIONS:
-                        r = parse_path_argument(optarg, false, &arg_definitions);
+                case ARG_DEFINITIONS: {
+                        _cleanup_free_ char *path = NULL;
+                        r = parse_path_argument(optarg, false, &path);
                         if (r < 0)
                                 return r;
+                        if (strv_extend(&arg_definitions, path) < 0)
+                                return log_oom();
```
Comment (poettering) [inline src/partition/repart.c:4288]: maybe also do strv_uniq() somewhere here

File: src/partition/repart.c [hunk 3]
```diff
@@ -4281,11 +4281,15 @@ static int parse_argv(int argc, char *argv[]) {
...
                         break;
 
-                case ARG_DEFINITIONS:
-                        r = parse_path_argument(optarg, false, &arg_definitions);
+                case ARG_DEFINITIONS: {
+                        _cleanup_free_ char *path = NULL;
+                        r = parse_path_argument(optarg, false, &path);
                         if (r < 0)
                                 return r;
+                        if (strv_extend(&arg_definitions, path) < 0)
```
Comment (poettering) [inline src/partition/repart.c]: use strv_push() and TAKE_PTR afterwards to donate the allocated memory to the strv. i.e.

```c
if (strv_push(&arg_definitions, &path) < 0)
        return log_oom();

TAKE_PTR(path);
```

File: src/partition/repart.c [hunk 4]
```diff
@@ -4904,7 +4908,7 @@ static int run(int argc, char *argv[]) {
         if (!context)
                 return log_oom();
 
-        r = context_read_definitions(context, arg_definitions, arg_root);
+        r = context_read_definitions(context, (const char * const*)arg_definitions, arg_root);
```
Comment (poettering) [inline src/partition/repart.c]: I'd probably just drop the 'const' from the context_read_definitions() parameter. constant arrays of strings are a mess, you have to cast all the time. Hence so far we said: don't bother with const if you deal with a `char**` `strv` array...

**Comment by poettering** [comment]
looks good, just some minor comments

File: src/partition/repart.c (hunk 3 above)
Comment (daandemeyer) [reply src/partition/repart.c]: What about 

```
if (strv_consume(&arg_definitions), TAKE_PTR(path)) < 0)
        return log_oom();
```

?

Comment (poettering) [reply src/partition/repart.c]: You are right, of course!

**Comment by poettering** [comment]
lgtm, but you marked the PR as "draft" still?

**Pull request merged**